### ISSUE TITLE
crates in private registries result in lots of warnings

### ISSUE THREAD

**Issue by jayvdb** [trailofbits/cargo-unmaintained#278]
We have workspace dependencies like

```toml
depA = { version = "1.2.3", registry = "our-private-registry" }
```

I see this warning repeated many times
```
warning: failed to get latest version of `my-private-crate`: failed to find `my-private-crate` in index
```

and then it is listed as ` (no repository)`, which is reasonable, but this message could be improved because it is known that the dep was not able to be followed.

**Comment by smoelius** [comment]
> but this message could be improved because it is known that the dep was not able to be followed.

So, to be sure I understand, the behavior you would prefer would be a more informative warning message? Would you like to propose one?

**Comment by jayvdb** [comment]
Most important is the tool should recognise that the dependency has specified a custom `registry = ".."` and should not attempt to find it in crates.io , as that is unnecessary network traffic, and is also leaking private crate names in the API queries going across public networks.

Instead of listing these dependency individually as "(no repository)" when they do have a repository, a note at the bottom of the output grouping them together would be nice.  Something like:

> Unable to follow these dependencies using a custom registry: dep_a, dep_b, etc.

**Comment by smoelius** [comment]
I think this issue should be resolved by #281. Could I impose on you to try out that branch?
```sh
cargo install cargo-unmaintained --git https://github.com/trailofbits/cargo-unmaintained --branch private-registry
```
